src/freenet/client/async/SplitFileInserterStorage.java

```diff
@@ -218,7 +223,16 @@
 
     private final int overallStatusLength;
     
+    /** Protects cooldown calculations and noBlocksToSend. You MUST NOT take this lock while 
+     * holding any other locks!!!
```

Comment:
> I don't get the impression that Fred itself is particularly memory-hungry - it always seems to be plugins or a huge queue causing problems.

… or a large data store (remember the 1/3000th rule of thumb for the bloom filters), or a large number of threads on 64-bit JVMs (~1MiB per thread), a lot of running requests (multi-tab browsing and image-heavy sites), and some minor effects from plenty of other areas.

Either way, I'm all for per-segment getters. The slightly greater memory usage is well worth the reduced code complexity* and locking IMHO. I guess this should be a separate discussion on the bugtracker instead of something blocking this pull request.

* not to mention the non-deterministic worse-than-linear algorithmic complexity of `chooseBlock()`, I'll have a fix incoming for that.